### ISSUE TITLE
Add error if a `Generator` not unpacked.

### ISSUE THREAD

**Issue by F-park** [microsoft/pyright#7087]
**Is your feature request related to a problem? Please describe.**

As we have `reportUnusedExpression`, I consider we can add `reportUnusedGenerator`

```py
def func():
    yield

func()  # should generate error because do nothing
```

**Describe the solution you’d like**

```py
def outer():
    def inner():
        yield 1
        yield 2

    inner()  # should generate error because do nothing
    return 1  # should generate error because need yield first and then return

    yield from inner()  # expected


outer()  # should generate error because do nothing

r = [*outer()]  # expected working in star operator
map(lambda x: x, outer())  # expected working in function which requires argument for Iterable
for _ in outer():  # expected working with for statement
    ...
```
